From b9e155a28bc802bfe3341e4ede7a8743879feb92 Mon Sep 17 00:00:00 2001
From: Dingxian Wen <shawn.wen@rock-chips.com>
Date: Fri, 7 Aug 2020 15:12:31 +0800
Subject: [PATCH] media: v4l2-core: add support for VIDIOC_G_INPUT ioctl on
 subdev nodes

Add VIDIOC_G_INPUT ioctl support to get the input status of subdev.

Signed-off-by: Dingxian Wen <shawn.wen@rock-chips.com>
Change-Id: I7d4253e058e17db7e2247efab17f639f4f3ef508
---
 drivers/media/v4l2-core/v4l2-subdev.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/media/v4l2-core/v4l2-subdev.c b/drivers/media/v4l2-core/v4l2-subdev.c
index 2b63fa6b6fc9..fd2ba87a0c5c 100644
--- a/drivers/media/v4l2-core/v4l2-subdev.c
+++ b/drivers/media/v4l2-core/v4l2-subdev.c
@@ -495,6 +495,9 @@ static long subdev_do_ioctl(struct file *file, unsigned int cmd, void *arg)
 	case VIDIOC_SUBDEV_S_DV_TIMINGS:
 		return v4l2_subdev_call(sd, video, s_dv_timings, arg);
 
+	case VIDIOC_G_INPUT:
+		return v4l2_subdev_call(sd, video, g_input_status, arg);
+
 	case VIDIOC_SUBDEV_G_STD:
 		return v4l2_subdev_call(sd, video, g_std, arg);
 
-- 
2.35.3

